File: framework/scala/src/main/scala/soda/scala/web/EchoHandler.scala
package soda.scala.web

import com.sun.net.httpserver.HttpExchange

import soda.scala.web.http.RequestHelper
import soda.scala.web.resp.{Response, ResponseFactory}

class EchoHandler extends BaseHandler {

  override protected def doGet(exchange: HttpExchange): Response = {
    ResponseFactory.success("echo " + RequestHelper.queryString(exchange))
  }

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/EchoHandler.kt
package soda.kotlin.web

import com.sun.net.httpserver.HttpExchange
import soda.kotlin.web.http.RequestHelper
import soda.kotlin.web.resp.Response
import soda.kotlin.web.resp.ResponseFactory

class EchoHandler : BaseHandler() {
    override fun doGet(exchange: HttpExchange?): Response {
        return ResponseFactory.success("echo " + RequestHelper.queryString(exchange!!))
    }
}